src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,115 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private final User firstValidUser = new User();
+    private final User secondValidUser = new User();
+    private final User thirdValidUser = new User();
+    private final User sameFirstUser = new User();
+    private final User nullAgeUser = new User();
+    private final User nullLoginUser = new User();
+    private final User nullPasswordUser = new User();
+    private final User negativeAgeUser = new User();
+    private final User incorrectPasswordUser = new User();
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        firstValidUser.setLogin("FirstUser");
+        firstValidUser.setAge(18);
+        firstValidUser.setPassword("Correct pass");
+        secondValidUser.setLogin("SecondUser");
+        secondValidUser.setAge(19);
+        secondValidUser.setPassword("Correct password");
+        thirdValidUser.setLogin("Third User");
+        thirdValidUser.setAge(20);
+        thirdValidUser.setPassword("Password");
+        sameFirstUser.setLogin("FirstUser");
+        sameFirstUser.setAge(18);
+        sameFirstUser.setPassword("Correct pass");
+        nullAgeUser.setLogin("Null age user");
+        nullAgeUser.setAge(null);
+        nullAgeUser.setPassword("Correct pass");
+        nullLoginUser.setLogin(null);
+        nullLoginUser.setAge(20);
+        nullLoginUser.setPassword("Correct pass");
+        nullPasswordUser.setLogin("Null password user");
+        nullPasswordUser.setAge(19);
+        nullPasswordUser.setPassword(null);
+        negativeAgeUser.setLogin("Negative age user");
+        negativeAgeUser.setAge(-19);
+        negativeAgeUser.setPassword("Correct Password");
+        incorrectPasswordUser.setLogin("Incorrect password user");
+        incorrectPasswordUser.setAge(20);
+        incorrectPasswordUser.setPassword("pass");
```

Comment:
Seems like you can reduce the amount of code, by using one `user` object with default values. Then you can just update fileds in the test methods when needed.